src/main/java/bio/terra/cli/utils/SAMUtils.java

```diff
@@ -0,0 +1,155 @@
+   */
+  public static SystemStatus getStatus(ApiClient apiClient) {
+    StatusApi statusApi = new StatusApi(apiClient);
+    SystemStatus status = null;
+    try {
+      status = statusApi.getSystemStatus();
+    } catch (ApiException apiEx) {
+      logger.error("Error getting SAM status", apiEx);
+    } finally {
+      closeConnectionPool(apiClient);
+    }
+    return status;
+  }
+
+  /**
+   * Call the SAM "/register/user/v2/self/info" endpoint to get the user info for the current user
+   * (i.e. the one whose credentials were supplied to the apiClient object).
+   *
+   * @return the SAM user status info object
+   */
+  public static UserStatusInfo getUserInfo(ApiClient apiClient) {
+    UsersApi samUsersApi = new UsersApi(apiClient);
+    UserStatusInfo userStatusInfo = null;
+    try {
+      userStatusInfo = samUsersApi.getUserStatusInfo();
+    } catch (ApiException apiEx) {
+      logger.error("Error getting user info from SAM.", apiEx);
+    } finally {
+      closeConnectionPool(apiClient);
+    }
+    return userStatusInfo;
+  }
+
+  /**
+   * Populate the terra user id and name properties of the given Terra user object.
+   *
+   * @param terraUser the Terra user whose credentials are supplied to the API client object
+   * @param globalContext the global context holds a pointer to the SAM instance
+   */
+  public static void populateTerraUserInfo(TerraUser terraUser, GlobalContext globalContext) {
+    ApiClient samClient = getClientForTerraUser(terraUser, globalContext.server);
+    UserStatusInfo samUserInfo = getUserInfo(samClient);
+    terraUser.terraUserId = samUserInfo.getUserSubjectId();
+    terraUser.terraUserName = samUserInfo.getUserEmail();
+  }
+
+  /**
+   * Try to close the connection pool after we're finished with this SAM request. TODO: why is this
+   * needed? possibly a bad interaction with picoCLI?
+   */
+  private static void closeConnectionPool(ApiClient apiClient) {
+    // try to close the connection pool after we're finished with this request -- why is this
+    // needed?
+    try {
+      apiClient.getHttpClient().connectionPool().evictAll();
+    } catch (Exception anyEx) {
+      logger.debug(
+          "Error forcing connection pool to shutdown after making a SAM client library call.",
+          anyEx);
+    }
+  }
+
+  /**
+   * Call the SAM "/api/google/v1/user/petServiceAccount/key" endpoint to get the pet SA key for the
+   * current user (i.e. the one whose credentials were supplied to the apiClient object).
+   *
+   * @return the HTTP response to the SAM request
+   */
+  public static HTTPUtils.HTTPResponse getPetSAKey(
+      TerraUser terraUser, GlobalContext globalContext) {
+    // The code below should be changed to use the SAM client library. For example:
+    //   GoogleApi samGoogleApi = new GoogleApi(apiClient);
+    //   samGoogleApi.getArbitraryPetServiceAccountKey();
+    // But I couldn't get this to work. The ApiClient throws an exception, I think in parsing the
```

Comment:
"I" is me. I didn't copy any code from anywhere else in Terra. The login stuff was copied from a BQ example provided by Google -- I didn't use TDR example there because it was using deprecated classes at the time (might have changed now, I haven't checked).

Agreed on following-up with CI team. I have another question for them about possibly whitelisting a client secret as well. Planning to do this after completing the command model.